src/deq_eid/main.py

```diff
@@ -0,0 +1,362 @@
+            self.gis,
+            self.secrets.INCIDENTS_ITEM_ID,
+            working_dir=self.tempdir_path,
+        )
+        incidents_count = incidents_loader.truncate_and_load(incidents_sdf)
+
+        chemical_df = self._get_chemical()
+        chemical_loader = load.ServiceUpdater(
+            self.gis,
+            self.secrets.CHEMICAL_ITEM_ID,
+            service_type="table",
+            working_dir=self.tempdir_path,
+        )
+        chemical_count = chemical_loader.truncate_and_load(chemical_df)
+
+        media_df = self._get_media()
+        media_loader = load.ServiceUpdater(
+            self.gis,
+            self.secrets.MEDIA_ITEM_ID,
+            service_type="table",
+            working_dir=self.tempdir_path,
+        )
+        media_count = media_loader.truncate_and_load(media_df)
+
+        end = datetime.now()
+
+        summary_message = MessageDetails()
+        summary_message.subject = f"{config.SKID_NAME} Update Summary"
+        summary_rows = [
+            f'{config.SKID_NAME} update {start.strftime("%Y-%m-%d")}',
+            "=" * 20,
+            "",
+            f'Start time: {start.strftime("%H:%M:%S")}',
+            f'End time: {end.strftime("%H:%M:%S")}',
+            f"Duration: {str(end-start)}",
+            "",
+            f"{config.INCIDENTS_TITLE} rows loaded: {incidents_count}",
+            f"{config.CHEMICAL_TITLE} rows loaded: {chemical_count}",
+            f"{config.MEDIA_TITLE} rows loaded: {media_count}",
+        ]
+
+        summary_message.message = "\n".join(summary_rows)
+        summary_message.attachments = self.tempdir_path / self.log_name
+
+        # print(summary_message.message)
+
+        self.supervisor.notify(summary_message)
+
+        self._remove_log_file_handlers()
+
+    def publish(self):
+        """Publish new AGOL hosted feature layers"""
+
+        #: NOTE: this method requires arcpy
+
+        incidents_sdf = self._get_incidents()
+        self._publish_dataset(
+            config.INCIDENTS_TABLE_NAME,
+            config.INCIDENTS_TITLE,
+            config.INCIDENTS_FIELDS,
+            incidents_sdf,
+            "layer",
+        )
+
+        chemical_df = self._get_chemical()
+        self._publish_dataset(
+            config.CHEMICAL_TABLE_NAME,
+            config.CHEMICAL_TITLE,
+            config.CHEMICAL_FIELDS,
+            chemical_df,
+            "table",
+        )
+
+        media_df = self._get_media()
+        self._publish_dataset(
+            config.MEDIA_TABLE_NAME,
+            config.MEDIA_TITLE,
+            config.MEDIA_FIELDS,
+            media_df,
+            "table",
+        )
+
+        self._remove_log_file_handlers()
+
+
+def process() -> None:
+    """Entry point triggered by the scheduler job
+
+    Returns:
+        None. The output is written to Cloud Logging.
+    """
+
+    skid = Skid()
+
+    #: choose one of the following
+    # skid.publish()  #: requires arcpy
+    skid.update()
```

Comment:
**suggestion**: it would be an improvement if there was a way to run both methods without having to modify this code to run it.

Reply:
I didn't feel like it was worth the effort for a method that will likely only ever be run once.